certora/WormholeOracleAuth.spec

```diff
@@ -0,0 +1,301 @@
+    postFeeAmount, totalFee = requestMint(e, guid, signatures, maxFeePercentage, operatorFee);
+
+    bytes32 sourceDomain;
+    bytes32 targetDomain;
+    bytes32 receiver;
+    bytes32 operator;
+    uint128 amount;
+    uint80 nonce;
+    uint48 timestamp;
+    sourceDomain, targetDomain, receiver, operator, amount, nonce, timestamp = join.wormholeGUID();
+    assert(sourceDomain == guid.sourceDomain, "guid.sourceDomain was not preserved");
+    assert(targetDomain == guid.targetDomain, "guid.targetDomain was not preserved");
+    assert(receiver == guid.receiver, "guid.receiver was not preserved");
+    assert(operator == guid.operator, "guid.operator was not preserved");
+    assert(amount == guid.amount, "guid.amount was not preserved");
+    assert(nonce == guid.nonce, "guid.nonce was not preserved");
+    assert(timestamp == guid.timestamp, "guid.timestamp was not preserved");
+    assert(join.maxFeePercentage() == maxFeePercentage, "maxFeePercentage was not preserved");
+    assert(join.operatorFee() == operatorFee, "operatorFee was not preserved");
+    assert(join.postFeeAmount() == postFeeAmount, "postFeeAmount was not preserved");
+    assert(join.totalFee() == totalFee, "totalFee was not preserved");
+}
+
+// Verify revert rules on requestMint
+rule requestMint_revert(
+        oracle.WormholeGUID guid,
+        bytes signatures,
+        uint256 maxFeePercentage,
+        uint256 operatorFee
+) {
+    env e;
+
+    require(guid.amount <= max_uint128);
+    require(guid.nonce <= 0xffffffffffffffffffff);
+    require(guid.timestamp <= 0xffffffffffff);
+
+    require(wormholeJoin() == join);
+    require(aux.oracle() == currentContract);
+
+    uint256 ward = wards(e.msg.sender);
+    address operatorAddr = aux.bytes32ToAddress(guid.operator);
+    uint256 threshold = threshold();
+    uint256 count = signatures.length / 65;
+    uint256 i;
+    require(i + 1 < count);
+    bytes32 hash = aux.getSignHash(guid);
+    uint256 vI;
+    bytes32 rI;
+    bytes32 sI;
+    vI, rI, sI = aux.splitSignature(signatures, i);
+    address recoveredI = aux.callEcrecover(hash, vI, rI, sI);
+    uint256 vIPlus1;
+    bytes32 rIPlus1;
+    bytes32 sIPlus1;
+    vIPlus1, rIPlus1, sIPlus1 = aux.splitSignature(signatures, i + 1);
+    address recoveredIPlus1 = aux.callEcrecover(hash, vIPlus1, rIPlus1, sIPlus1);
+
+    uint256 numProcessedBeforeI;
+    uint256 numValidBeforeI;
+    numProcessedBeforeI, numValidBeforeI = aux.processUpToIndex(hash, signatures, i);
+    uint256 numProcessedBeforeIPlus1;
+    uint256 numValidBeforeIPlus1;
+    numProcessedBeforeIPlus1, numValidBeforeIPlus1 = aux.processUpToIndex(hash, signatures, i + 1);
+
+    uint256 a;
+    uint256 numValid;
+    a, numValid = aux.processUpToIndex(hash, signatures, count);
+
+    requestMint@withrevert(e, guid, signatures, maxFeePercentage, operatorFee);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = ward != 1;
+    bool revert3 = e.msg.sender != operatorAddr;
```

Comment:
I don't understand why the proof accepts with these conditions. As far as I can tell, `wards` is not checked by `WormholeOracleAuth.requestMint`, and I think `revert3` really out to be `operator != msg.sender && receiver != msg.sender`.